pkg/ibmvpc/parser.go

```diff
@@ -557,90 +585,95 @@ func getFipConfig(
 	return nil
 }
 
-func isVPCToAnalyzeConfigured(vpcID string) bool {
-	return vpcID != ""
-}
-
-func getVPCconfig(rc *ResourcesContainer, res *vpcmodel.CloudConfig, vpcID string, skipByVPC func(string) bool) (*VPC, error) {
-	if len(rc.vpcsList) > 1 && !isVPCToAnalyzeConfigured(vpcID) {
-		return nil, fmt.Errorf("supporting currently only one vpc configuration")
-	}
-	for i := range rc.vpcsList {
-		vpc := rc.vpcsList[i]
+func getVPCconfig(rc *ResourcesContainer, res map[string]*vpcmodel.CloudConfig, skipByVPC func(string) bool) error {
+	for _, vpc := range rc.vpcsList {
 		if skipByVPC(*vpc.CRN) {
 			continue // skip vpc not specified to analyze
 		}
+		vpcName := *vpc.Name
 		vpcNodeSet := &VPC{
-			VPCResource: vpcmodel.VPCResource{ResourceName: *vpc.Name, ResourceUID: *vpc.CRN, ResourceType: ResourceTypeVPC},
+			VPCResource: vpcmodel.VPCResource{ResourceName: vpcName, ResourceUID: *vpc.CRN, ResourceType: ResourceTypeVPC},
 			nodes:       []vpcmodel.Node{},
 			zones:       map[string]*Zone{},
 		}
-		res.NodeSets = append(res.NodeSets, vpcNodeSet)
-		res.NameToResource[vpcNodeSet.Name()] = vpcNodeSet
-		// assuming one vpc
-		return vpcNodeSet, nil
+		res[vpcName] = NewEmptyCloudConfig()
+		res[vpcName].NodeSets = append(res[vpcName].NodeSets, vpcNodeSet)
+		res[vpcName].NameToResource[vpcNodeSet.Name()] = vpcNodeSet
 	}
-	return nil, errors.New("could not find a VPC config")
+	if len(res) == 0 {
+		return errors.New("could not find any VPC to analyze")
+	}
+	return nil
 }
 
-func singleVPCErr(vpcName, layerVPCName string) error {
-	return fmt.Errorf("NACL/SG VPC (%s) is different from its layer's VPC (%s)", vpcName, layerVPCName)
+func parseSGTargets(sgResource *SecurityGroup,
+	sg *vpc1.SecurityGroup,
+	res map[string]*vpcmodel.CloudConfig,
+	intfNameToIntf map[string]*NetworkInterface,
+	vpcName string) {
```

Comment:
`res` and `vpcName` only appear as `res[vpcName]`.
Maybe only pass a pointer to the relevant `CloudConfig`.

Reply:
done